### PR TITLE
task: Make Feature refresher register as client

### PR DESCRIPTION
**Pull request by chriswk**
When a new token for registration comes in, feature refresher registers as a client upstream

We've discussed this before. After talking to Gard and looking at what we were using client registrations for, it made sense for Edge to register as a client for Client tokens that it's fetching data for.

### CODE REVIEW

**Review by gardleopard — APPROVED** [review]
Fantastic!

**Review by sighphyre — APPROVED** [review]
LGTM!

File: server/src/http/feature_refresher.rs [hunk 1]
```diff
@@ -26,6 +28,19 @@ pub struct FeatureRefresher {
     pub persistence: Option<Arc<dyn EdgePersistence>>,
 }
 
+fn client_application_from_token(token: EdgeToken) -> ClientApplication {
+    ClientApplication {
+        app_name: "unleash_edge".into(),
+        connect_via: None,
+        environment: token.environment,
+        instance_id: None,
+        interval: 5,
```
Comment (sighphyre) [inline server/src/http/feature_refresher.rs:37]: Should we include the actual polling interval here? Might be useful

Comment (chriswk) [reply server/src/http/feature_refresher.rs:37]: Agreed. I'll fix

**Pull request merged**